jhotdraw-core/src/main/java/org/jhotdraw/draw/figure/BezierFigure.java

```diff
@@ -184,24 +185,23 @@ public boolean contains(Point2D.Double p, double scaleDenominator) {
               AttributeKeys.getStrokeTotalWidth(this, scaleDenominator)
                   * attr().get(STROKE_MITER_LIMIT));
       if (gs.createStrokedShape(path).contains(p)) {
-        return true;
+        result = true;
       } else {
         if (isClosed()) {
           return false;
         }
       }
-    }
-    if (!isClosed()) {
+    } else if (!isClosed()) {
```

Comment:
these have a lot if else statements that are nested within one another which results more lines of code but can be implemented in one - 
example -  (!isClosed()) and else if (!isClosed()) can be combined into one. i see a repeatable lines of code here